src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvoker.cs

```diff
@@ -386,6 +391,32 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
             await result.ExecuteResultAsync(_pageContext);
         }
 
+        private async Task<object[]> GetArguments(HandlerMethodDescriptor handler)
+        {
+            var arguments = new object[handler.Parameters.Length];
+            var valueProvider = await CompositeValueProvider.CreateAsync(_pageContext, _pageContext.ValueProviderFactories);
+
+            for (var i = 0; i < handler.Parameters.Length; i++)
+            {
+                var parameter = handler.Parameters[i];
+                var parameterDescriptor = new ParameterDescriptor
+                {
+                    Name = parameter.Name,
+                    ParameterType = parameter.Type,
+                };
+
+                var result = await _parameterBinder.BindModelAsync(
+                    _page.PageContext,
+                    valueProvider,
+                    parameterDescriptor,
+                    value: null);
```

Comment:
This is actually not good, this is going to lead to memory leaks. 

We need to expose a new method on `ParameterBinder`.